core/resolver.go

```diff
@@ -236,3 +225,33 @@ func MoveSource(recipe *Recipe, source Source) error {
 		return fmt.Errorf("unsupported source type %s", source.Type)
 	}
 }
+
+// checksumValidation validates the checksum of a file
+func checksumValidation(source Source, path string) error {
+	//No checksum provided
+	if source.Checksum == "" {
```

Comment:
`len(strings.TrimSpace(s)) == 0` may be a more suited check here since the module may contain `"  "` as the checksum, which would not return true in this check